api/graphql/schema.graphql

```diff
@@ -513,6 +513,11 @@ input PoolFilterInput {
     id: ID!
 }
 
+input UserPoolFilterInput {
+    ids: [ID!]!
+    viewExpiredCandidates: Boolean
```

Comment:
It seems strange to me that the parameter is added here.  Why isn't it on PoolCandidateFilterInput like `isWoman` and the other booleans?

Reply:
Oh I see, the concept of being expired only exists in the context of a specific pool.  I wonder if we could name this better in case someone else is confused like me.  

Naming is hard!  Maybe we can change `ids` to `poolIds` since this isn't an input just for `pool` anymore.

Maybe `isPoolCandidateExpired` for the other parameter is a little more consistent with the other fields?

It's a little weird that this filter works differently from the other boolean filters.  For example, "is_woman = false" is the same as "is_woman is null".  In both cases you get all candidates.  For this filter, "viewExpiredCandidates = false" gets you all the actives only, right?  Do we want to change this to match the behavior of the other filters?  Return everybody unless the filter exists and is set to true?  I guess I would call it `includeOnlyActive` then?

I think I'm going to ask Tristan for a quick design meetup today in scrum.

Reply:
So I've changed `ids` to `pools` for UserPoolFilter input but I still don't like it because it leaves us with queries like this:
```
'where' => [
    'pools' => [
        'pools' => [$myPool->id],
        'expiredStatus' => ApiEnums::CANDIDATE_EXPIRY_FILTER_ACTIVE
    ]
]
```
I'm considering renaming the outer `pools` to something a little more accurate but I'm not sure what - `poolsAndStatus` maybe. Any other thoughts on naming improvements?

Also I've added the `CANDIDATE_EXPIRY_FILTER` enum and updated the user query with it. I'm currently switching the Pool Candidate queries to use it as well.

Reply:
Ok, Pool Candidate queries are changed over to use the `CANDIDATE_EXPIRY_FILTER` enum as well.

Reply:
Yeah, I agree, that query is ugly with the two pools.  I also agree that changing the outer input is the way to go.  `PoolAndStatus` would become inaccurate as soon as we add anything else.. Maybe `poolCandidates`?  These are all filters in the pool_candidate table context, right?

Reply:
I noticed a weird quirk while manually testing.  It seems like it's returning the filtered list of users but the paginator is pulling in everyone.  Is the paginator bypassing the contexts?
![image](https://user-images.githubusercontent.com/8978655/170713833-5c5d53b6-443e-4deb-b230-e925c2209c8f.png)

Reply:
I played around a bit myself but haven't been able to replicate your bug. Refreshing my environment didn't change anything. Would you mind taking another look and seeing if it's still happening for you?

Also in case it helps:
- This is the paginated query which defaults to only 10 results per page so that might be what made you think the list of users was filtered.
- The Pool Candidate factory currently only generates valid users so unless you change things the `ACTIVE` and `ALL`  statuses will both return all users in the pool while `EXPIRED` will not return anyone unless you change things.

Reply:
My attempt

![image](https://user-images.githubusercontent.com/40485260/170794397-941b2c81-8543-4268-b5aa-113e4a16c0ba.png)

Reply:
Yup, there is a default to 10 

![image](https://user-images.githubusercontent.com/40485260/170794514-2c3d5ec1-1a22-405e-b4b9-a7a7540f4a7f.png)

Reply:
> My attempt

That looks right. The seeder generates 71 users, 2 of which are pool managers and the rest are active in one of those two pools.

Reply:
So, the trouble here is that our seeder only seeds active pool candidates. :cry: 
```
select expiry_status, count(*) from (
  select
    case
      when expiry_date > current_date then 'active'
      when expiry_date is null then 'null'
      when expiry_date <= current_date then 'expired'
    end as expiry_status
  from pool_candidates pc
  where pool_id = '1a81dc03-32da-45be-9b49-d4d7da996058'
) t
group by expiry_status
```
![image](https://user-images.githubusercontent.com/8978655/171037643-777d9dd8-c3e8-4f8e-89ae-3b70a6d7b5d8.png)
I created #2894 to improve that.
So I manually fixed that in my workspace.  Here's a gnarly update statement:
```
UPDATE pool_candidates
SET expiry_date=u.new_expiry_date
FROM (
  select
    id,
    case
      when row_num < 20 then '1970-01-01'
      when row_num < 40 then null
      else expiry_date
    end as new_expiry_date
  from (
    select
      id,
      expiry_date,
      row_number() over (order by id) row_num 
    from pool_candidates pc
  ) as t
) AS u
WHERE pool_candidates.id=u.id;
```
That looks better:
![image](https://user-images.githubusercontent.com/8978655/171037716-8aa4f7a6-9f37-402f-98d5-3bad8746e276.png)
Unfortunately, I'm seeing everyone in the paginator, all statuses and pools.  Also, the paginator doesn't seem to be limiting page size or page, either?
![image](https://user-images.githubusercontent.com/8978655/171038843-e502bf2e-5a28-4684-9aeb-e7cfddddf227.png)

```
query q($where: UserFilterAndOrderInput, $first: Int, $page: Int) {
  usersPaginated(where: $where, first: $first, page: $page) {
    data {
      firstName
    }
    paginatorInfo {
      count
      currentPage
      hasMorePages
      perPage
      total
    }
  }
}
```
```
{
  "where": {
    "poolCandidates": {
      "pools": ["1a81dc03-32da-45be-9b49-d4d7da996058"],
      "expiryStatus": "ACTIVE"
    },
  },
  "first": 3,
  "page": 2
}
```